README.md

```diff
@@ -1,30 +1,135 @@
 # PostCSS Customprop Validate
 
-[PostCSS] plugin to validate default values of CSS custom properties.
+[PostCSS] plugin to validate fallback values of CSS custom properties.
 
-[PostCSS]: https://github.com/postcss/postcss
+Validating the fallback values of CSS custom properties manually can be error prone and challening if they are huge in number. This plugin removes this effort altogether.
+
+The plugin accepts an object containing custom properties in plugin [options]. Using it as the source of truth, the plugin validates the CSS and finds out the custom properties with incorrect fallback values.
+
+The plugin can also modify the source file with the updated CSS -- with correct fallback values. See `write` under [options]
+
+> If you use a code formatter like [prettier], you may need to re-run it after the plugin modifies the source file.
+
+## Options
+
+The plugin accepts 3 options:
+
+**`properties`**
+
+An object containg key and fallback values of custom properties
+
+| type   | required | default |
+| ------ | -------- | ------- |
+| Object | true     | N/A     |
+
+**`write`**
+
+Writes the source CSS file with the updated CSS
+
+| type    | required | default |
+| ------- | -------- | ------- |
+| boolean | false    | false   |
+
+**`callback`**
+
+Callback to handle the response of the plugin. The callback accepts 2 arguments:
+
+- error: an error object
+- data: the response
+
+| type     | required | default  |
+| -------- | -------- | -------- |
+| Function | false    | () => {} |
+
+## Response
+
+The response is a list of object of custom properties with incorrect fallback values. The plugin's result also contains the response -- `result.wrongProps`. Shape of the object:
```

Comment:
* Avoid `--`. If you meant to use an endash – or an emdash —, use that directly. https://en.wikipedia.org/wiki/Dash
* What is the difference between a result and a response?
* Does it makes sense to say something like: `The response is an array of objects. Each object represents an incorrect fallback value having the following shape.` In JavaScript, list is an array, so better said that way. Other programming languages have different data types to represent various collections & lists.

Reply:
result is the common terminology used to represent the result which a plugin produces after processing the CSS. Response is a term that I am using to represent the argument of the callback.

P.S. I had this doubt whether it will confuse the readers. But I tried to explain it, hence created a section altogether.